lib/Slackbot_worker.js

```diff
@@ -86,10 +105,22 @@ module.exports = function(botkit, config) {
             bot.rtm = new Ws(res.url, null, {agent: agent});
             bot.msgcount = 1;
 
-            var pingIntervalId = null;
+            bot.rtm.on('pong', function (obj) {
+                lastPong = Date.now()
+                botkit.debug('PONG received');
+            })
+
             bot.rtm.on('open', function() {
 
                 pingIntervalId = setInterval(function() {
+                    if (lastPong && lastPong + 12000 < Date.now()) {
```

Comment:
Slack's RTM websocket connection didn't support ping/pong in the past; however, it does today. I tested this change quite a bit with up to 40 socket connections at a time and various failure scenarios.